jax/interpreters/batching.py

```diff
@@ -275,9 +276,24 @@ def batch_custom_vjp_bwd(bwd, axis_name, axis_size, in_dims, out_dim_dests, main
 def _match_axes_and_sum(axis_size, out_dims_thunk, out_dim_dests, *in_vals):
   # this is like _match_axes, but we do reduce-sums as needed
   out_vals = yield in_vals, {}
-  yield map(partial(matchaxis, axis_size, sum_match=True),
+  yield map(partial(_matchaxis_symbolic_zeros, axis_size, sum_match=True),
             out_dims_thunk(), out_dim_dests, out_vals)
 
+def _matchaxis_symbolic_zeros(sz, src, dst, x, sum_match=False):
```

Comment:
This change is unrelated to the bug at hand, but it is related to producing symbolic Zeros as the output of a custom_vjp bwd function (mentioned above). The reason we weren't producing symbolic zeros before is that we had to make this function, which is only called in the vmap rule for custom_vjp, handle them.